pilot/cmd/pilot-agent/main.go

```diff
@@ -435,7 +435,7 @@ var (
 
 			// If control plane auth is not mTLS or global SDS flag is turned off, unset UDS path and token path
 			// for control plane SDS.
-			if !controlPlaneAuthEnabled || !nodeAgentSDSEnabled {
+			if !nodeAgentSDSEnabled {
```

Comment:
Update the comment above your change as it's misleading.

I think control plane auth is needed to do SDS else you will be transporting certs over plain-text.

Reply:
Its a bit orthogonal, isn't it? SDS is now the only support mechanism to serve certs to Envoy, there is no file mounts. So we should either get rid of the plaintext port entirely or make it work?

Reply:
I'm with you on that. Basically the summary is:

* If SDS is enabled, control plane security must be turned on and non-secure ports blocked.
* If control plane security is disabled for whatever reason, I want to not use SDS and if file mounts are not configured, I would rather have traffic not working vs leaking certs.

Reply:
Ok then lets just kill the plaintext port then

Reply:
Works for me. What are the implications for users:

* Does any istioctl command relies on reaching Pilot via unsecure port?

Reply:
Its just XDS, istioctl connects over a different unsecure port 8080

cc @costinm